addons/mass_mailing/models/mailing_list.py

```diff
@@ -223,6 +225,24 @@ def action_merge(self, src_lists, archive):
     def close_dialog(self):
         return {'type': 'ir.actions.act_window_close'}
 
+    # ------------------------------------------------------
+    # MAILING
+    # ------------------------------------------------------
+
+    def _mailing_get_default_domain(self, mailing):
+        return [('list_ids', 'in', mailing.contact_list_ids.ids)]
+
+    def _mailing_get_opt_out_list(self, mailing):
+        """ Check subscription on all involved mailing lists. If user is opt_out
+        on one list but not on another if two users with same email address, one
+        opted in and the other one opted out, send the mail anyway. """
+        # TODO DBE Fixme : Optimize the following to get real opt_out and opt_in
+        subscriptions = self.subscription_ids if self else mailing.contact_list_ids.subscription_ids
+        opt_out_contacts = subscriptions.filtered(lambda rel: rel.opt_out).mapped('contact_id.email_normalized')
+        opt_in_contacts = subscriptions.filtered(lambda rel: not rel.opt_out).mapped('contact_id.email_normalized')
+        opt_out = set(c for c in opt_out_contacts if c not in opt_in_contacts)
```

Comment:
Wow... 
I guess the same email_address can be in both opt_out or opt_in in different list yes.
Shouldn't opt_out take precedence over opt_in for "safety" reasons?

Reply:
Currently it is the inverse ... one opt-in wons. Unsubscribe links are included to manage your subscriptions, you may completely blacklist (wons over opt-out), and unsubscribe will even be improved soon. So I will keep it as it is currently to avoid functional changes :) .